### PR TITLE
Support for Docker container-based development for ease and consistency

### PR DESCRIPTION
**Pull request by fsufitch**

### CODE REVIEW

**Review by fragglet — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+FROM ubuntu
+
+######### Apt dependencies
+
+ARG system_apt_dependencies="build-essential automake autoconf libtool git pkg-config python3 python3-pip sudo"
+ARG choco_apt_dependencies="gcc make libsdl2-dev libsdl2-net-dev libsdl2-mixer-dev"
```
[the remark this answers is not in the document]
Comment (fsufitch) [reply .devcontainer/Dockerfile:6]: Absolutely. Does Choco gracefully downgrade if it's built with libpng and libpng isn't available later, though?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.209.6/containers/docker-existing-dockerfile
+{
+	"name": "Existing Dockerfile",
```
[the remark this answers is not in the document]
Comment (fsufitch) [reply .devcontainer/devcontainer.json:4]: Yeah, it should, this was just the default. I'm doing a few more polish passes and removing placeholders, pointless "guide" comments and making values more significant/specific is part of it.

[the remark this answers is not in the document]
Comment (fsufitch) [reply .devcontainer/devcontainer.json]: Fixed it.

File: .devcontainer/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (fsufitch) [reply .devcontainer/Dockerfile:6]: Fixed.

**Review by fragglet — APPROVED** [review]

**Pull request merged**